### PR TITLE
Do not run `stubtest_third_party` on requirements change

### PR DESCRIPTION
**Pull request by sobolevn**
Similar to https://github.com/python/typeshed/blob/main/.github/workflows/stubtest_stdlib.yml

### CODE REVIEW

**Comment by sobolevn** [comment]
Right now on `requirements-tests.txt` change it will run all the machinery, find 0 stubs change and exit with `0` code.

File: .github/workflows/stubtest_third_party.yml [hunk 1]
```diff
@@ -5,6 +5,8 @@ on:
     paths-ignore:
       - '**/*.md'
       - 'scripts/**'
+      # When requirements.txt changes, we run `daily.yml`, which includes third-party stubtest
+      - 'requirements-tests.txt'
```
Comment (AlexWaygood) [inline .github/workflows/stubtest_third_party.yml:9]: Why don't we just do this, now that (following #8923), this job now has its own workflow file?

```suggestion
    paths:
      - 'stubs/**'
      - '.github/workflows/stubtest_third_party.yml'
      - 'tests/**'
      # When requirements.txt changes, we run `daily.yml`, which includes third-party stubtest
```

**Comment by sobolevn** [comment]
Yes, this is even better!

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Pull request merged**